src/components/checklist.jsx

```diff
@@ -0,0 +1,115 @@
+import { useState } from "react";
+
+function Checklist() {
+  const [tasks, setTasks] = useState([]);
+  const [inputValue, setInputValue] = useState("");
+  const [editTask, setEditTask] = useState(false);
+  const [currTaskIndex, setCurrTaskIndex] = useState(null);
+  const [isAddTaskCompleted, setIsAddTaskCompleted] = useState(false);
+
+  // Handle input change.
+  const handleInputChange = (event) => {
+    setInputValue(event.target.value);
+  };
+
+  // Allows users to add a new task.
+  const handleAddTask = (event) => {
+    event.preventDefault();
+    // Only accepts inputs that are not empty.
+    if (inputValue.trim() !== "") {
+      setTasks([...tasks, inputValue]);
+      setInputValue("");
+    }
+  };
+
+  // Allows the selected task to be edited.
+  const handleEditTask = (index) => {
+    setInputValue(tasks[index]);
+    setEditTask(true);
+    setCurrTaskIndex(index);
+  };
+
+  // Allows the task to be edit and input updated.
+  const handleUpdateTask = (event) => {
+    event.preventDefault();
+    if (inputValue.trim() !== "") {
+      const taskList = [...tasks];
+      //Update the selected task with new input.
+      taskList[currTaskIndex] = inputValue;
+      setTasks(taskList);
+      setEditTask(false);
+      setCurrTaskIndex(null);
+      setInputValue("");
+    }
+  };
+
+  // Allows users to remove a task.
+  const handleRemoveTask = (index) => {
+    const taskList = tasks.filter((_, i) => i !== index);
+    setTasks(taskList);
+  };
+
+  //Allow users to choose between adding more tasks or be done with adding tasks.
+  const handleCompleteAdding = () => {
+    setIsAddTaskCompleted(true);
+  };
+
+  const handleContAdding = () => {
```

Comment:
I think the function name here is unclear. We should be more specific. Not sure what `Cont` is